scripts/paper/phase44b_public_projection_logup_probe.py

```
@@ -0,0 +1,1066 @@
+        "stwo_air_proof_claimed": False,
+        "trace_commitment": "",
+    }
+    trace["appended_pairs_commitment"] = commit_phase43_trace_appended_pairs(trace)
+    trace["lookup_rows_commitments_commitment"] = commit_phase43_trace_lookup_rows_commitments(trace)
+    trace["phase30_step_envelopes_commitment"] = commit_phase43_trace_phase30_step_envelopes(trace)
+    trace["phase30_source_chain_commitment"] = commit_phase43_trace_source_chain(trace)
+    trace["trace_commitment"] = commit_phase43_history_replay_trace(trace)
+    return trace
+
+
+def verify_demo_trace(trace: dict[str, Any]) -> None:
```

Ruff (0.15.10)

[warning] Too many branches (23 > 12)

(PLR0912)